rocAL/rocAL_pybind/amd/rocal/plugin/tf.py

```diff
@@ -50,8 +50,18 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
         color_format = b.getOutputColorFormat(self.loader._handle)
         self.p = (1 if (color_format == int(types.GRAY)) else 3)
 
-        self.out = np.zeros(( self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype = "uint8")
-
+        if self.tensor_dtype == types.FLOAT:
```

Comment:
please add variable for dtype and use it in the code instead of 2 separate conditions